server/sonar-server/src/test/java/org/sonar/server/ui/ws/GlobalNavigationActionTest.java

```diff
@@ -0,0 +1,169 @@
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.ui.ws;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.sonar.api.config.Settings;
+import org.sonar.api.utils.System2;
+import org.sonar.api.web.Page;
+import org.sonar.api.web.View;
+import org.sonar.core.dashboard.ActiveDashboardDao;
+import org.sonar.core.dashboard.ActiveDashboardDto;
+import org.sonar.core.dashboard.DashboardDao;
+import org.sonar.core.dashboard.DashboardDto;
+import org.sonar.core.persistence.DbSession;
+import org.sonar.core.persistence.DbTester;
+import org.sonar.core.user.UserDto;
+import org.sonar.server.db.DbClient;
+import org.sonar.server.ui.Views;
+import org.sonar.server.user.MockUserSession;
+import org.sonar.server.user.db.UserDao;
+import org.sonar.server.ws.WsTester;
+
+import static org.mockito.Mockito.mock;
+
+public class GlobalNavigationActionTest {
+
+  @ClassRule
+  public static final DbTester dbTester = new DbTester();
+
+  private DbSession session;
+
+  private WsTester wsTester;
+
+  private UserDao userDao;
+
+  private DashboardDao dashboardDao;
+
+  private ActiveDashboardDao activeDashboardDao;
+
+  @Before
+  public void before() throws Exception {
+    dbTester.truncateTables();
+
+    userDao = new UserDao(dbTester.myBatis(), mock(System2.class));
+    dashboardDao = new DashboardDao(dbTester.myBatis());
+    activeDashboardDao = new ActiveDashboardDao(dbTester.myBatis());
+    DbClient dbClient = new DbClient(
+      dbTester.database(), dbTester.myBatis(), userDao, dashboardDao, activeDashboardDao);
+
+    session = dbClient.openSession(false);
+  }
+
+  @After
+  public void after() throws Exception {
+    session.close();
+  }
+
+  @Test
+  public void empty_call() throws Exception {
+    wsTester = new WsTester(new NavigationWs(new GlobalNavigationAction(activeDashboardDao, new Views(), new Settings())));
+
+    MockUserSession.set();
+
+    wsTester.newGetRequest("api/navigation", "global").execute().assertJson(getClass(), "empty.json");
+  }
+
+  @Test
+  public void nominal_call_for_anonymous() throws Exception {
+    nominalSetup();
```

Comment:
Good point ! It this case, to make it clear to the reader, I would use 2 methods : createDashboardForAnonymous() and createDashboardForUser() (Or something like that)